Terraform.gitignore

```diff
@@ -35,3 +35,22 @@ override.tf.json
 # Ignore CLI configuration files
 .terraformrc
 terraform.rc
+
+# Ignore hcl files
+.terraform.lock.hcl
+*.terraform.*
+
+# Ignore graph files
+*.dot
+
+# to skip MacOS custom attributes. It appears when creating a .bicep file.
+.DS_Store
+
+# Ignore plan files (optional) as some might save plan before destroy the terraform configuration
```

Comment:
If this is optional (per the comment) let's just comment out the line and let folks uncomment if they need to